src/trees/avl/utils.ts

```diff
@@ -0,0 +1,52 @@
+import type { AVLNode } from './node.js';
+
+export const BALANCE_STATE = {
```

Comment:
Why can't you use the same values you use in `getBalanceFactor`?

Reply:
had to refactor a bit, but I agree let me sort this out :)